src/Common/src/Interop/Linux/procfs/Interop.ProcFsStat.cs

```diff
@@ -285,5 +302,73 @@ internal static bool TryParseStatFile(string statFilePath, out ParsedStat result
             result = results;
             return true;
         }
+
+        internal static bool TryReadStatusFile(int pid, out ParsedStatus result, ReusableTextReader reusableReader)
+        {
+            bool b = TryParseStatusFile(GetStatusFilePathForProcess(pid), out result, reusableReader);
+            Debug.Assert(!b || result.pid == pid, "Expected process ID from status file to match supplied pid");
+            return b;
+        }
```

Comment:
Does some other code call TrParseStatusFile?  I'm wondering what the reasoning was for separating it out from this TryReadStatusFile method, which just layers on a single GetStatusFilePathForProcess call.

Reply:
I followed the same format as for `TryParseStatFile`/`TryReadStatFile`.

Reply:
> I followed the same format as for TryParseStatFile/TryReadStatFile.

Understood.  That one's a bit different though in that TryReadStatFile has multiple call sites.

Reply:
`TryParseStatFile` is called directly by a test (`src/Common/tests/Tests/Interop/procfsTests.cs`) so if we are to add tests for this `TryParseStatusFile` (which I am sure you expect), then there is a point in separating them.

Reply:
I see. Ok.